core/src/Cuda/Kokkos_Cuda_Parallel_Range.hpp

```diff
@@ -421,7 +421,8 @@ class ParallelScan<FunctorType, Kokkos::RangePolicy<Traits...>, Kokkos::Cuda> {
   //  (c) gridDim.x  <= blockDim.y * blockDim.y
   //  (d) gridDim.y  == gridDim.z == 1
 
-  const FunctorType m_functor;
+  const CombinedFunctorReducer<FunctorType, typename Analysis::Reducer>
+      m_functor_reducer;
```

Comment:
`m_functor_reducer` gets created on every invocation of `parallel_scan` on the host (just like `m_functor` before). This member is effectively memcopied to the device. The point of this pull request is that the `FunctorAnalysis` object is not created on the device (since that invokes the functor's constructor and thus also the destructor eventually). Before the changes to the `ParallelReduce` interface, `FunctorAnalysis` was storing a reference so calling its constructor and destructor wasn't a problem (since it didn't invoke the functor's constructor and destructor).

Reply:
I see. Thanks for explaining that.